File: go/p009.go
package main

import "fmt"

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	for a := 1; a < 500; a++ {
		low := max(a, 500-a)
		for b := low; b < 500; b++ {
			c := 1000 - a - b
			if a*a+b*b == c*c {
				fmt.Println(a * b * c)
				// output: 31875000
			}
		}
	}
}

File: rust/p009.rs
fn main() {
    for a in range(1, 500) {
        let low = std::num::max(a, 500-a);
        for b in range(low, 500) {
            let c = 1000 - a - b;
            if a*a+b*b == c*c {
                println!("{}", a * b * c)
                // output: 31875000
            }
        }
    }
}
